opm/material/common/IntervalTabulated2DFunction.hpp

```diff
@@ -158,83 +135,125 @@ class XYTabulated2DFunction
     Scalar valueAt(size_t i, size_t j) const
     { return samples_[i][j]; }
 
+    /*!
+     * \brief Returns true if a coordinate lies in the tabulated range
+     */
+    template <class Evaluation>
```

Comment:
since the extrapolation for the two direction might be different, I am not when we need this function. 

typically the usage will always involves the extrapolation together, like the following 
```
if ((!xExtrapolate_ && !appliesX(x)) || (!yExtrapolate_ && !appliesY(y)))
```

Reply:
Yes, this method is not strictly necessary, but it is more convenient than what you propose and also it unifies the API with the other tabulation classes. the intended use case is something like this:

```c++
if (fn.applies(x, y))
  // no extrapoltion required. do something with the function value
else
  // do something funky
```

I'm pretty sure that this required/useful for the ECL use case.